packages/legacy/core/App/screens/Settings.tsx

```diff
@@ -102,8 +102,15 @@ const Settings: React.FC<SettingsProps> = ({ navigation }) => {
   const settingsSections: SettingSection[] = [
     {
       header: {
-        icon: 'apartment',
-        title: t('Screens.Contacts'),
+        icon: { name: store.preferences.useConnectionInviterCapability ? 'person' : 'apartment', size: 30 },
+        title: store.preferences.useConnectionInviterCapability ? store.preferences.walletName : t('Screens.Contacts'),
+        iconRight: {
```

Comment:
I am misreading this or is the edit icon not gated?

Reply:
it is gated in the JSX template: https://github.com/hyperledger/aries-mobile-agent-react-native/blob/7be0b09278392c174b99091b0d3cef6e7c022df4/packages/legacy/core/App/screens/Settings.tsx#L254